frontend/pkg/frontend/frontend.go

```diff
@@ -147,6 +147,40 @@ func (f *Frontend) ArmResourceListBySubscription(writer http.ResponseWriter, req
 
 	f.logger.Info(fmt.Sprintf("%s: ArmResourceListBySubscription", versionedInterface))
 
+	subscriptionId := request.PathValue(PathSegmentSubscriptionID)
+
+	query := fmt.Sprintf("azure.subscription_id='%s'", subscriptionId)
+	clustersListResponse, err := f.conn.ClustersMgmt().V1().Clusters().List().Search(query).Send()
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	systemData := &arm.SystemData{} //Empty data until https://issues.redhat.com/browse/ARO-8002 is completed
+	var hcpCluster *api.HCPOpenShiftCluster
+	clusters := clustersListResponse.Items().Slice()
+	for _, cluster := range clusters {
+
+		hcpCluster, err = f.ConvertCStoHCPOpenShiftCluster(systemData, cluster)
+		if err != nil {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+
+		versionedResource := versionedInterface.NewHCPOpenShiftCluster(hcpCluster)
+		resp, err := json.Marshal(versionedResource)
+		if err != nil {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+		_, err = writer.Write(resp)
```

Comment:
Not sure, just a question off the top of my head: should we create a slice of clusters and then marshal + write the single variable containing all of them? Not sure if this returns a valid JSON or some valid JSONs concatenated, which results in not a valid JSON.

(Same comment for the other functions)

Reply:
Output is looking good in the slice:
` curl -X GET "localhost:8443/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.RedHatOpenshift/hcpOpenShiftClusters?api-version=2024-06-10-preview"
{"Value":[{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg/providers/Microsoft.RedHatOpenShift/hcpOpenShiftClusters/klin-test-name","location":"eastus","name":"klin-test-name","properties":{"provisioningState":"","spec":{"api":{"url":"","visibility":"external"},"console":{"url":""},"disableUserWorkloadMonitoring":false,"dns":{"baseDomain":"","baseDomainPrefix":"klin-aro-hcp"},"etcdEncryption":false,"externalAuth":{"enabled":false,"externalAuths":[]},"fips":false,"ingress":[{"ip":"","url":"","visibility":""}],"issuerUrl":"","network":{"hostPrefix":23,"machineCidr":"10.0.0.0/16","networkType":"OVNKubernetes","podCidr":"10.128.0.0/14","serviceCidr":"172.30.0.0/16"},"platform":{"etcdEncryptionSetId":"","managedResourceGroup":"klin-test-mrg","networkSecurityGroupId":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg/providers/Microsoft.Network/networkSecurityGroups/myNSG","outboundType":"loadBalancer","subnetId":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg/providers/Microsoft.Network/virtualNetworks/vnet/subnets/mySubnet"},"proxy":{"httpProxy":"","httpsProxy":"","noProxy":"","trustedCa":""},"version":{"availableUpgrades":[],"channelGroup":"stable","id":"openshift-v4.15.11"}}},"systemData":{"createdBy":"","createdByType":"","lastModifiedBy":"","lastModifiedByType":""},"tags":{},"type":"Microsoft.RedHatOpenShift/hcpOpenShiftClusters"},{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg-2/providers/Microsoft.RedHatOpenShift/hcpOpenShiftClusters/klin-test-name-2","location":"eastus","name":"klin-test-name-2","properties":{"provisioningState":"","spec":{"api":{"url":"","visibility":"external"},"console":{"url":""},"disableUserWorkloadMonitoring":false,"dns":{"baseDomain":"","baseDomainPrefix":"klin-aro-hcp-2"},"etcdEncryption":false,"externalAuth":{"enabled":false,"externalAuths":[]},"fips":false,"ingress":[{"ip":"","url":"","visibility":""}],"issuerUrl":"","network":{"hostPrefix":23,"machineCidr":"10.0.0.0/16","networkType":"OVNKubernetes","podCidr":"10.128.0.0/14","serviceCidr":"172.30.0.0/16"},"platform":{"etcdEncryptionSetId":"","managedResourceGroup":"klin-test-mrg","networkSecurityGroupId":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg-2/providers/Microsoft.Network/networkSecurityGroups/myNSG","outboundType":"loadBalancer","subnetId":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/klin-test-rg-2/providers/Microsoft.Network/virtualNetworks/vnet/subnets/mySubnet"},"proxy":{"httpProxy":"","httpsProxy":"","noProxy":"","trustedCa":""},"version":{"availableUpgrades":[],"channelGroup":"stable","id":"openshift-v4.15.11"}}},"systemData":{"createdBy":"","createdByType":"","lastModifiedBy":"","lastModifiedByType":""},"tags":{},"type":"Microsoft.RedHatOpenShift/hcpOpenShiftClusters"}],"NextLink":null}%                       `